### PR TITLE
feat: Next.js template

### PR DESCRIPTION
**Pull request by andipaetzold**
Adding a Next.js template

### CODE REVIEW

File: examples/nextjs/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
+    "test": "jest --watch",
+    "test:ci": "jest --ci"
+  },
```
Comment (loweisz) [inline examples/nextjs/package.json:12]: Shall we include the `"create-app-definition"` `"upload"` and `"upload-ci"` as we have in the other templates?

File: examples/nextjs/pages/locations/ConfigScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React, { useCallback, useState, useEffect } from 'react';
```
Comment (loweisz) [inline examples/nextjs/pages/locations/ConfigScreen.tsx:1]: Why are all components under `/pages/locations`? wouldn't that mean theoretically there is now a route to `/pages/locations/ConfigScreen` ?

File: examples/nextjs/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+This project was bootstrapped with [Create Contentful App](https://github.com/contentful/create-contentful-app).
+
+## Config Location
+
+Due to a bug in Next.js the config location doesn't load in development mode. The bug has been reported in Next.js' GitHub repository: https://github.com/vercel/next.js/issues/18028.
```
Comment (loweisz) [inline examples/nextjs/README.md:5]:
```suggestion
Due to a bug in Next.js, the config location doesn't load in development mode. The bug has been reported in Next.js' GitHub repository: https://github.com/vercel/next.js/issues/18028.
```

File: examples/nextjs/package.json (hunk 1 above)
Comment (andipaetzold) [reply examples/nextjs/package.json:12]: Won't work here. This is SSR and can't be hosted statically

File: examples/nextjs/pages/locations/ConfigScreen.tsx (hunk 2 above)
Comment (andipaetzold) [reply examples/nextjs/pages/locations/ConfigScreen.tsx:1]: I don't understand? This is just structured the same as the React template.

File: examples/nextjs/package.json (hunk 1 above)
Comment (loweisz) [reply examples/nextjs/package.json:12]: True, but the `create-app-definition` will work or not?

File: examples/nextjs/pages/locations/ConfigScreen.tsx (hunk 2 above)
Comment (loweisz) [reply examples/nextjs/pages/locations/ConfigScreen.tsx:1]: In the other templates we only have `src/locations/<location>.tsx`, but here it is `src/pages/locations/<location>.tsx`, I just wonder why we need that `pages` folder as we don't have routing

Comment (andipaetzold) [reply examples/nextjs/pages/locations/ConfigScreen.tsx:1]: tbh, I have no experience about Next.js and just put the files into the folder that was created via create-next

Comment (loweisz) [reply examples/nextjs/pages/locations/ConfigScreen.tsx:1]: The idea is that if you put things in the `pages` folder  automatically creates the route like that.
https://nextjs.org/docs/basic-features/pages

That means we technically now have routes like `/pages/locations/ConfigScreen` available which we don't need. 
So having the same structure as the other js/ts templates should be working here as well

Comment (andipaetzold) [reply examples/nextjs/pages/locations/ConfigScreen.tsx:1]: Okay, gonna move it. Thanks

File: examples/nextjs/package.json (hunk 1 above)
Comment (andipaetzold) [reply examples/nextjs/package.json:13]: yes, added that

File: examples/nextjs/pages/locations/ConfigScreen.tsx (hunk 2 above)
Comment (andipaetzold) [reply examples/nextjs/pages/locations/ConfigScreen.tsx]: done

**Review by loweisz — APPROVED** [review]
🔥

**Pull request merged**